kythe/go/serving/pipeline/beam.go

```diff
@@ -596,12 +642,24 @@ func nodeToEdges(n *scpb.Node, emit func(*spb.VName, *scpb.Edge)) {
 	}
 }
 
+func nodeWithoutEdges(n *scpb.Node) *scpb.Node {
+	return &scpb.Node{
+		Source:  n.Source,
+		Kind:    n.Kind,
+		Subkind: n.Subkind,
+		Fact:    n.Fact,
+	}
+}
+
 // reverseEdge emits the reverse of each *scpb.Edge, embedding the associated TargetNode.
 func reverseEdge(src *spb.VName, nodeStream func(**scpb.Node) bool, edgeStream func(**scpb.Edge) bool, emit func(*spb.VName, *scpb.Edge)) {
 	var node *scpb.Node
 	if !nodeStream(&node) {
-		node = &scpb.Node{Source: src}
+		node = &scpb.Node{}
+	} else {
+		node = nodeWithoutEdges(node)
 	}
+	node.Source = src
```

Comment:
why do we always set node.Source now?

Reply:
Nodes passed through this will not have their `Source` set since it was pulled out in the input `KV`'s key.